src/btree/row_srch.c

```diff
@@ -492,16 +483,11 @@ __wt_row_search(WT_CURSOR_BTREE *cbt, WT_ITEM *srch_key, bool insert, WT_REF *le
             }
         else if (collator == NULL) {
             /*
-             * Reset the skipped prefix counts; we'd normally expect the parent's skipped prefix
-             * values to be larger than the child's values and so we'd only increase them as we walk
-             * down the tree (in other words, if we can skip N bytes on the parent, we can skip at
-             * least N bytes on the child). However, if a child internal page was split up into the
-             * parent, the child page's key space will have been truncated, and the values from the
-             * parent's search may be wrong for the child. We only need to reset the high count
-             * because the split-page algorithm truncates the end of the internal page's key space,
-             * the low count is still correct.
+             * In some cases we expect we're comparing more than a few keys with matching prefixes,
+             * so it's faster to avoid the memory fetches by skipping over those prefixes. That's
+             * done by tracking the length of the prefix match for the lowest and highest keys.
              */
-            skiphigh = 0;
+            skiphigh = skiplow = 0;
```

Comment:
Can you mention in this comment that `skiphigh` and `skiplow` are always set to 0 when we look at a page? Removing the default initialization of `skiplow` up above was throwing me off.
Same comment where we set these variables on line 641

Reply:
It's worth keeping context in the comment regarding why they must be reset across page traversals. I'd hate to add a similar optimization again in the future if we forgot.

I also asked a question in the Jira ticket about this solution to make sure I fully understand the issue.

Reply:
@agorrod @quchenhao Sid and I have discussed this comment and are happy with the current iteration.

@quchenhao we've dropped "ancestor page" in favour of using parent and its child which I think conveys the same information about the relation to grandparent pages, but feel free to request ancestor again if you think it's clearer.
@agorrod we've changed the wording about why we need to reset skiphigh/skiplow but I think kept the intention behind the comment. Again feel free to request the original comment if it's clearer.